sdk/go/common/apitype/core.go

```diff
@@ -287,6 +287,8 @@ type ResourceV3 struct {
 	Aliases []resource.URN `json:"aliases,omitempty" yaml:"aliases,omitempty"`
 	// CustomTimeouts is a configuration block that can be used to control timeouts of CRUD operations
 	CustomTimeouts *resource.CustomTimeouts `json:"customTimeouts,omitempty" yaml:"customTimeouts,omitempty"`
+	// ImportID is the import input used for imported resources.
+	ImportID resource.ID `json:"importID,omitempty" yaml:"importID,omitempty"`
```

Comment:
We may want to bump the statefile version: if a user updates their stack with a version of the CLI that stores the import ID, then updates again with an older version, the second update will either lose the import ID (if it succeeds), or fail due to a replace (i.e. the situation this bug fixes). Personally I'm in favor of bumping the version for this, esp. since it aligns with 2.0.

Reply:
Should we just bump `DeploymentSchemaVersionCurrent` from `3` to `4` and add `4` to the following case here (along with some comments):

https://github.com/pulumi/pulumi/blob/77969c83e50f557534ee1773572ce659730eb002/pkg/resource/stack/deployment.go#L167-L170

Or would it be more clear to go through the ceremony of adding a `ResourceV4`, adding a migration (a migration that does nothing?)?

Reply:
> Or would it be more clear to go through the ceremony of adding a ResourceV4, adding a migration (a migration that does nothing?)?

I think this would be clearer.

Reply:
I am going to merge this without the statefile version bump. https://github.com/pulumi/pulumi/issues/4409 tracks following-up post 2.0 launch.